tests/baselines/reference/inferTypes1.types

```diff
@@ -121,12 +121,12 @@ type U15 = InstanceType<typeof Abstract>;  // Abstract
 >Abstract : typeof Abstract
 
 type U16<T extends any[]> = InstanceType<new (x: string, ...args: T) => T[]>;  // T[]
->U16 : T[]
+>U16 : U16<T>
 >x : string
 >args : T
 
 type U17<T extends any[]> = InstanceType<abstract new (x: string, ...args: T) => T[]>;  // T[]
->U17 : T[]
+>U17 : U17<T>

```

Comment:
Those 3 changes here are basically caused by the same issue. 

Because `target`/`source` is now `any[]` and not `any` when we start comparing the signatures within the conditional type we don't pass the `isTypeAssignable` check [here](https://github.dev/microsoft/TypeScript/blob/a179e910dd818262571b46313a7918aacf67978a/src/compiler/checker.ts#L16141). Because of that the conditional type can't be "simplified" here to the `trueType` and thus the displayed type changed.

This check reaches the `structuredTypeRelatedToWorker` with types like:
```
source // any[]
target // [x: string, ...args: T]
```

I've expected this to be satisfied somewhere around [this check](https://github.dev/microsoft/TypeScript/blob/a179e910dd818262571b46313a7918aacf67978a/src/compiler/checker.ts#L19871) but actually the `target` is not an array - it's a tuple and there is no logic supporting tuple targets here. The question is if this is deliberate or just an accidental omission?

I've played around with this more though and I've noticed 2 additional problems here though:
1. `any[]` is **not** assignable to `[string, ...any[]]`. I understand why the error is reported for this case but in the presence of `any` in the type system it feels weird. Intuitively I've expected `any[]` to be assignable to all arrays and tuples, similarly to how `anyFunctionType` is always assignable, regardless of the parameters count etc, see the check [here](https://github.dev/microsoft/TypeScript/blob/a179e910dd818262571b46313a7918aacf67978a/src/compiler/checker.ts#L20406-L20408)
2. even if the previous point would get "fixed" then the base constraint for `T` here is unknown and not `any[]`. I don't understand why as I've expected the latter.